src/core/namespace.c

```diff
@@ -202,7 +204,17 @@ static const MountEntry protect_home_yes_table[] = {
 
 /* ProtectControlGroups=yes table */
 static const MountEntry protect_control_groups_yes_table[] = {
-        { "/sys/fs/cgroup",      MOUNT_READ_ONLY,     true  },
+        { "/sys/fs/cgroup",      MOUNT_READ_ONLY,                   false  },
+};
+
+/* ProtectControlGroups=private table */
+static const MountEntry protect_control_groups_private_table[] = {
+        { "/sys/fs/cgroup",      MOUNT_PRIVATE_CGROUP2FS,           false  },
+};
+
+/* ProtectControlGroups=strict table */
+static const MountEntry protect_control_groups_strict_table[] = {
+        { "/sys/fs/cgroup",      MOUNT_PRIVATE_CGROUP2FS_READ_ONLY, false  },
```

Comment:
Maybe, better to reuse `MOUNT_PRIVATE_CGROUP2FS`, but set `.readonly = true` ?

Reply:
Yea so I tried that but it turns out append_static_mounts ignores the read_only field: https://github.com/systemd/systemd/blob/2f6fe4e1131d39fcafa9e00a7902919efb5361e1/src/core/namespace.c#L718-L720

Likely because all the static mounts do not declared as global variables don't initialize it. Let me try fixing that so we can properly pass read_only...will probably put that as a separate commit in the stack since we will have to change all static mount global vars

If we do that, we could also get rid of MOUNT_PRIVATE_TMP_READ_ONLY too

Reply:
Actually I'm not gonna change this in this PR because I actually found a bug with ProtectHome=tmpfs and potentially /run: https://github.com/systemd/systemd/issues/34825

I think we should refactor and fix the bug in another PR probably

Reply:
After #34834 is merged, `MOUNT_PRIVATE_CGROUP2FS` can be reused with `.read_only = true`.